infrastructure/sentry-subgraph/src/utils/updateChallenge.ts

```diff
@@ -38,6 +39,15 @@ export function updateChallenge(referee: Referee, challenge: Challenge): Challen
     challenge.numberOfEligibleClaimers = challengeStruct.numberOfEligibleClaimers;
     challenge.amountClaimedByClaimers = challengeStruct.amountClaimedByClaimers;
 
+    //set challenge field in NodeConfirmed event to challenge id and save
+    let nodeConfirmation = NodeConfirmation.load(challenge.assertionId.toString());
+    if (!nodeConfirmation) {
+        log.warning(`Failed to load nodeConfirmation entity with id: ${challenge.assertionId}`, []);
+        return challenge;
+    }
+    nodeConfirmation.challenge = challenge.id.toString();
+    nodeConfirmation.save();
```

Comment:
This needs to happen in a loop for all NodeConfirmed assertionIds if this is a bundled challenge for multiple assertionIds.

Reply:
Added the loop based on assertion id gap